lib/rig/application.ex

```diff
@@ -2,17 +2,32 @@ defmodule Rig.Application do
   @moduledoc false
 
   use Application
-  use Rig.Config, [:log_level]
+  use Rig.Config, [:log_level, :log_type]
 
   alias RigOutboundGateway.Kinesis
   alias RigOutboundGateway.KinesisFirehose
 
+  alias LoggerJSON.Formatters.BasicLogger
+
   def start(_type, _args) do
     alias Supervisor.Spec
 
     # Override application logging with environment variable
     Logger.configure([{:level, config().log_level}])
 
+    case config().log_type do
+      :json ->
+        Logger.add_backend(LoggerJSON)
+      :gcl ->
+        Logger.add_backend(LoggerJSON)
+        Logger.configure_backend(LoggerJSON, formatter: LoggerJSON.Formatters.GoogleCloudLogger)
+      _ ->
+        # TODO: Fix format; context: config.exs:94
+        # Or can I leave it like that? The only difference seems to
+        # be that the level and the module are switched
+        Logger.add_backend(:console)
```

Comment:
So the config for console is not used here? You could try to set the values using Application.put_env, which should basically do the same..

It would be good to respect the config setting if possible, because it deduplicates the meta data.

Reply:
What do you mean? I meant that the format specified in config.exs doesn't apply here. I can fix metadata just by setting `:metadata` to `:all`. I don't think I can set the format through an env var.

Reply:
I wasn't talking about environment variables - see https://hexdocs.pm/elixir/Application.html#put_env/4

Reply:
> I meant that the format specified in config.exs doesn't apply here. I can fix metadata just by setting :metadata to :all.

Why can't you apply the previous config here? Not sure what you mean..

Reply:
Hmm...it appears to work now...for some reason...

Reply:
Like...it produces the same result (log-wise) as the master branch

Reply:
@kevinbader Is this closable then?

Reply:
please double check whether it _actually_ works now ;) but yes, if it _really_ works, sure